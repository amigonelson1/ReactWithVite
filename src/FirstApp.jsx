import PropTypes from 'prop-types'; //Este importe es para capturar y devolver errores en las props types

//***** si no dependen del funcional component se recomienda dejar por fuera*****/

const titulo = 'Título'; // constantes fuera del funcional componente evita sobrecarga de lectura
const arreglo = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]; //se pueden pasar arreglos, pero NO objetos
const objeto = {
    nombre: 'Pepe',
    edad: 35,
};// para poder imprimir debemos usar JSON.stringify(objeto)

const suma = () => {
    return 2 + 3;
};

function FirstApp({ titulo, valor }) {

    return (
        <>
            <h1>first App, {titulo} con valor de {valor}</h1>
            <p>Este es un párrafo.</p>
            <p>{arreglo}</p>
            <code>{JSON.stringify(objeto)}</code>
            <p>{suma()}</p>
        </>
    )
};

//A continuacion se le asignan los tipos de valores y si son requeridos, en caso de no recibir alguno mostrara error en consola;
// se debe usar el nombre del componenete .propTypes;
FirstApp.propTypes = {
    titulo: PropTypes.string.isRequired,
    valor: PropTypes.number.isRequired,
};

//forma de asignar default props en el componenete es .defaultProps
FirstApp.defaultProps = {
    titulo: 'Soy el título por defecto',
    valor: 1,
    nombre: 'soy el nombre por defecto',
}

export default FirstApp