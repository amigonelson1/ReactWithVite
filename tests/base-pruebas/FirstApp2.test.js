import { render, screen } from '@testing-library/react';
import FirstApp from '../../src/FirstApp';
//'https://testing-library.com/docs/react-testing-library/intro' complemento para hacer testing a componentes de react;

describe('prueba a FistApp', () => {

    const titulo = 'Hola, soy un titulo';
    const valor = 222;

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp titulo={titulo} />);
        expect(container).toMatchSnapshot();
    });
    test('debe mostrar mensaje: Hola, soy un titulo', () => {
        render(<FirstApp titulo={titulo} />);
        expect(screen.getAllByText(titulo)).toBeTruthy();
    });
    test('mostrar el titulo en un h1', () => {
        render(<FirstApp titulo={titulo} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(titulo);
    });
    test('mostrar prop: valor n veces', () => {
        render(<FirstApp valor={valor} />);
        expect(screen.getAllByText(valor).length).toBe(3);
    });
});