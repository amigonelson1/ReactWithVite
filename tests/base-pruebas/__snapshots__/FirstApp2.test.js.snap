// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba a FistApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy un titulo
  </h1>
  <div
    data-testid="test-titulo"
  >
    Hola, soy un titulo
  </div>
  <p>
    1
  </p>
  <p>
    1
  </p>
  <p>
    1
  </p>
  <p>
    Este es un p√°rrafo.
  </p>
  <p>
    0
    1
    2
    3
    4
    5
    6
    7
    8
    9
  </p>
  <code>
    {"nombre":"Pepe","edad":35}
  </code>
  <p>
    5
  </p>
</div>
`;
